### DISCUSSION TITLE
Thread pool overload?

### DISCUSSION THREAD

**Discussion by xMaxximum** [quartznet/quartznet#2130] [Q&A] [answered]
Hello,

From reading the docs I got a question. Quartz utilises some sort of thread pool for concurrent job execution, correct? What would happen if the threadpool limit is at 10 but you have scheduled 11 jobs at the same time. Would the last one just get aborted and ignored or would it run after each one is finished? For my usecase the *exact* time doesn't matter but I just need to run them when they occur as each execution doesn't take too long

**Comment by asulwer** [comment]
this is a basic programming question, not really suited for here as it is not specific to Quartz.

the quick answer is that its Queued or should be.

https://www.csharptutorial.net/csharp-concurrency/csharp-threadpool/
https://owlcation.com/stem/C-ThreadPool-and-its-Task-Queue-Example

included some links for implementation of the threadpool and how to queue, not exhaustive.

**Comment by lahma** [comment, marked as answer]
The last job which didn't fit the 10 allowed would have to wait. If the wait is long enough misfire instructions come to play. Every job will be handled, sooner or later. If your setup is always creating more jobs than there are runners all the time that would of course cause exhaustion of resources.

**Comment by xMaxximum** [comment]
my bad for that rather simple question then, I never worked with a task scheduling library like this before and the docs somehow made it seem like the scenario I explained would pose a problem.
Thanks for both of your responses!
